tests/integration_tests/functional/test_logging.py

```diff
@@ -116,11 +116,44 @@ def test_error_logs(test_microvm_with_ssh):
     )
 
 
+def test_dirty_page_metrics(test_microvm_with_api):
+    """Check the `dirty_pages` metric."""
+    microvm = test_microvm_with_api
+    microvm.spawn()
+    microvm.basic_config()
+
+    # Configure logging.
+    log_fifo_path = os.path.join(microvm.path, 'log_fifo')
+    metrics_fifo_path = os.path.join(microvm.path, 'metrics_fifo')
+    log_fifo = log_tools.Fifo(log_fifo_path)
+    metrics_fifo = log_tools.Fifo(metrics_fifo_path)
+
+    response = microvm.logger.put(
+        log_fifo=microvm.create_jailed_resource(log_fifo.path),
+        metrics_fifo=microvm.create_jailed_resource(metrics_fifo.path),
+        level='Error',
+        show_level=False,
+        show_log_origin=False,
+        options=['LogDirtyPages']
+    )
+    assert microvm.api_session.is_good_response(response.status_code)
+
+    microvm.start()
+
+    lines = metrics_fifo.sequential_fifo_reader(3)
+    for line in lines:
+        assert int(json.loads(line)['memory']['dirty_pages']) >= 0
+        # TODO force metrics flushing and get real data without waiting for
```

Comment:
#642 would have helped here for sure...

Reply:
I had it in mind all the time while writing this test :smile: Customers wouldn't use it, but it would have come in super handy here, and future tests might depend on metrics too, so it would be nice to have for testing. `#[feature]` seems like overkill but I haven't thought of any nice clean way of having that functionality available just for integration tests yet :disappointed: